Add documentation for `order_by` (#631)
qdrant/landing_page: qdrant-landing/content/documentation/concepts/points.md

For convenience, this ID is also returned in the field `next_page_offset`.
If the value of the `next_page_offset` field is `null` - the last page is reached.

### Order points by a payload key

_Available as of v1.8_

> When using the [`scroll`](#scroll-points) API, it is possible to sort the results by the contents of a payload field. This is useful when you want to retrieve points in a specific order, for example, by a timestamp, for chronological order.

Author: Rephrased it a little different

> <aside role="status">Without an appropriate index, custom ordering would be a very expensive operation. For this reason, in Qdrant, it is a requirement to have an index capable of <a href=/documentation/concepts/indexing/#payload-index target="_blank">Range filtering conditions</a> on the field you want to `order_by`. Once the index is set up, these ordering operations can be really cheap.</aside>

Reviewer: By "expensive," I'm guessing you mean as in load on the system (not actual cost). I therefore suggest:

```suggestion
<aside role="status">Without an appropriate index, custom ordering can create costly loads. Qdrant therefore requires an index which supports <a href=/documentation/concepts/indexing/#payload-index target="_blank">Range filtering conditions</a> on one ore more specific `order_by` fields.</aside>
```

Author: "costly" is not the same as "expensive"?

Reviewer: > "costly" is not the same as "expensive"?

I asked ChatGPT the question, and I like the answer:

![Screenshot from 2024-02-22 12-25-53](https://github.com/qdrant/landing_page/assets/3287976/7eb945cd-1840-4c3c-969b-254201a56543)

```http
POST /collections/{collection_name}/points/scroll
{
    "limit": 15,
    "order_by": "timestamp", // <-- this!
}
```

```python
client.scroll(
    collection_name="{collection_name}",
    limit=15,
    order_by="timestamp", # <-- this!
)
```

```typescript
client.scroll("{collection_name}", {
  limit: 15,
  order_by: "timestamp", // <-- this!
});
```

```rust
use qdrant_client::qdrant::{Condition, Filter, ScrollPoints, OrderBy};

client
    .scroll(&ScrollPoints {
        collection_name: "{collection_name}".to_string(),
        limit: Some(15),
        order_by: Some(OrderBy {
            key: "timestamp".to_string(),  // <-- this!
            ..Default::default(),
        }),
        ..Default::default()
    })
    .await?;
```

> The `order_by` `key` parameter specifies the payload key to order the results by, but there are also other fields that can be set to control the ordering, like `direction` and `start_from`:

Reviewer: ```suggestion
The `order_by` `key` parameter is one way to order the results. You can substitute other parameters for `key` such as `direction` and `start_from`:
```

Reviewer: I don't feel strongly about this, will approve whether you commit this suggestion or not.

Author: Yeah, I think the suggestion _suggests_ that the `key` param is optional, but it is not. What I wanted to convey is that you can specify these other extra parameters along with the `key`

Reviewer: Ah thank you for explaining. Then maybe:

```suggestion
Use the `order_by` `key` parameter to specify the payload key. You can then add other fields to control the ordering, like `direction` and `start_from`:
```

```http
"order_by": {
    "key": "timestamp",
    "direction": "desc" // default is "asc"
    "start_from": 123, // start from this value
}
```

```python
order_by=models.OrderBy(
    key="timestamp",
    direction="desc",  # default is "asc"
    start_from=123,  # start from this value
)
```

```typescript
order_by: {
    key: "timestamp",
    direction: "desc", // default is "asc"
    start_from: 123, // start from this value
}
```

```rust
order_by: Some(OrderBy {
    key: "timestamp".to_string(),
    direction: Some(Direction::Desc) // default is Direction::Asc
    start_from: Some(123),
})
```

> Be mindful that, for array-like payloads, each point can appear as many times as the number of elements in the array. For example, if you have a point payload with a `timestamp` key, and the value for the key is an array of 3 elements, the same point will appear 3 times in the results, one for each timestamp.

Reviewer: ```suggestion
Note: for payloads organized in arrays, you may get what seems like duplicate results. For example, if you have a point payload with a `timestamp` key, and the value is an array of 3 elements, the same point appears 3 times in the results.
```

Author: Rephrased it a little different

> <aside role="alert">Pagination is disabled when using the `order_by` parameter. </aside>

Reviewer: ```suggestion
<aside role="alert">When you use the `order_by` parameter, pagination is disabled. </aside>
```

> As you may expect, pagination does not work exactly the same way than when the order is an ID, which is unique. Right now, pagination is not explicitly enabled when ordering by a payload key, so the `next_page_offset` field will not be present in the response. However, there is a trick you can do to achieve pagination:

Reviewer: ```suggestion
Pagination works differently when  `order_by` is an ID. Since pagination is not explicitly enabled, you do not see the `next_page_offset` field in the response. However, you can use the following steps to set up pagination:
```

> 1. For each page, store the last value of the `order_by` field.
> 2. Accumulate all ids which have the same "last value" in their `order_by` `key`.
> 3. Make a `has_id` filter with all the ids you have accumulated.
> 4. Use this filter and the "last value" as the `start_from` in the next request.

Reviewer: Tip: As long as you've set up spaces correctly, you don't have to specify consecutive numbers. This makes life easier if  you have to insert / delete a step.

```suggestion
1. For each page, store the last value of the `order_by` field.
1. Accumulate all IDs with the same "last value" in their `order_by` `key`.
1. Create a `has_id` filter with accumulated IDs.
1. Use this filter and the "last value" as the `start_from` in the next request.
```

Author: Today I learned about this, thanks!

For convenience, this ID is also returned in the field `next_page_offset`.
If the value of the `next_page_offset` field is `null` - the last page is reached.

### Order points by a payload key

> _Available as of v1.8_

Reviewer 2: We did specify patch version for older releases, e.g.
https://github.com/qdrant/landing_page/blob/069857d0c307abfaba8e4a3ae4a300c1c2004e11/qdrant-landing/content/documentation/concepts/points.md#L818

```suggestion
_Available as of v1.8.0_
```
